hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/HoodieTimelineArchiver.java

```diff
@@ -455,6 +458,10 @@ private Stream<HoodieInstant> getCommitInstantsToArchive() {
             }
             return true;
           }).filter(s ->
+              earlestUnCleanCompletedInstant.map(instant ->
+                      compareTimestamps(instant.getTimestamp(), GREATER_THAN, s.getTimestamp()))
+                  .orElse(true)
+          ).filter(s ->
```

Comment:
We better check the timeline clean metadata files, only when all the replaced clustering files are cleaned, can the clustering instant be archived.

Reply:
> We better check the timeline clean metadata files, only when all the replaced clustering files are cleaned, can the clustering instant be archived.


no completed clustering instant ->  no need to check
firstCompletedClusteringInstant is not null && no completed clean instant -> return firstCompletedClusteringInstant
firstCompletedClusteringInstant is not null && earliestCleanInstant is not null -> return firstUncleanClusteringInstant